### ISSUE TITLE
No such file or directory:"keys-th-{:0.1f}.npy"

### ISSUE THREAD

**Issue by InstantWindy** [ubc-vision/image-matching-benchmark#22]
Hi. If I run run.py on a test set, an error occurs when computing matches because there are no "keys-th-{:0.1f}.npy" files. So If I want to test it on my own data, how to generate "keys-th-{:0.1f}.npy" file? thanks.

**Comment by etrulls** [comment]
You cannot run experiments on the test set because you don't have the ground truth. That will be only the first of many errors. We will not share the ground truth for the test data, because it's part of a public challenge. If you want to compute results on the test set, please send a challenge submission (check the website for instructions).

**Comment by InstantWindy** [comment]
Thanks. If I just want to visualize the matching results after feature extraction, what should I do.

**Comment by etrulls** [comment]
They will be automatically uploaded to the website (a subset of it). Please check any entry on the leaderboards for examples.

**Comment by InstantWindy** [comment]
Sorry. I mean that I just want to visualize the matching results of the validation dataset,after training the model. I don't need to compare it with ground truth.

**Comment by etrulls** [comment]
Those will be dumped automatically into ../benchmark-visualization. If you want more images, you can play with the parameters.

**Comment by InstantWindy** [comment]
Yeah, I know.  But by run.py, it not only visualizes the results but also calculates match, filer, and so on. But I don't have the ground truth. I can't calculate the match. I just want to visualize the results.

**Comment by ducha-aiki** [comment]
You do have ground truth for validation set, download it from all_sequences.
![image](https://user-images.githubusercontent.com/4803565/81170432-f8cad780-8f9a-11ea-8ec2-23abe18cbdb9.png)

https://vision.uvic.ca/imw-challenge/index.md

**Comment by InstantWindy** [comment]
@ducha-aiki . Do I have to have groundtruth to visualize the match results?

**Comment by ducha-aiki** [comment]
Yes, because our match visualization script colors matches conditionally on them being wrong or correct:
![image](https://user-images.githubusercontent.com/4803565/81173423-51e93a00-8fa0-11ea-9ca1-8c197064bb6c.png)

Of course, you can hack into the script and remove that dependence, but in those case you need to do it yourself.
